### DISCUSSION TITLE
Mantine Charts

### DISCUSSION THREAD

**Discussion by Manas-Nagelia** [mantinedev/mantine#3074] [Feature requests]
Hey,

I love Mantine a lot, and it has become my favorite UI library.

I have an idea that the Mantine team could implement a separate package or a component for charts and graphs. This component/library could include multiple graph and chart types (line, bar, pie, mix of two, etc). This could be similar to Recharts, but it would support Mantine styling options.

Thanks!

**Comment by Garden-Playhouse** [comment]
I would also love to have this feature!

**Comment by x1yizhuo** [comment]
+1 I’d love to see this

**Comment by dukesx** [comment]
You can use Tremor for easing charts workflow , but keep in mind, dark mode is neither planned nor available.

**Reply by cseitz** [reply]
I looked into [Tremor](https://www.tremor.so/) and, although I have some major gripes with it and it is by no means fully-featured, it seems like a great basic charting library to mix with mantine!

They're apparently working on dark mode now. The fact that they didn't plan for dark mode from the beginning is surprising... but luckily you can just overwrite their theme with CSS.

Here are my global CSS rules that I use in Mantine to support dark mode for Tremor charts.

```tsx
import { Global, useMantineTheme } from '@mantine/core';

// Shorter cleaner version to say `check ? value : undefined` when looking at dozens of conditionals.
export function onlyIf<T>(check: boolean, value: T) {
    if (check) return value;
    return undefined;
}

// Just plop this somewhere on the page or app that renders tremor charts.
function TremorStyles() {
    const theme = useMantineTheme();
    const dark = theme.colorScheme === 'dark';
    const colors = theme.colors;

    return <Global styles={{
        '.recharts-wrapper .recharts-surface text': {
            fill: onlyIf(dark, colors.gray[0]),
        },
        '.recharts-cartesian-axis-tick text': {
            fill: onlyIf(dark, colors.gray[4] + '!important'),
        },
        '.tr-text-gray-500': {
            color: onlyIf(dark, colors.gray[4]),
        },
        '.tr-text-gray-700': {
            color: onlyIf(dark, colors.gray[0]),
        },
        '.tr-bg-white': {
            backgroundColor: onlyIf(dark, colors.dark[7]),
            color: onlyIf(dark, 'white'),
        },
        '.tr-ring-gray-200': {
            borderColor: onlyIf(dark, colors.dark[4]),
            '--tw-ring-color': onlyIf(dark, colors.dark[4]),
        },
        '.tremor-base :is(*, ::before, ::after)': {
            borderColor: onlyIf(dark, colors.dark[4]),
        },
        '.recharts-pie-sector path.recharts-sector': {
            stroke: onlyIf(dark, colors.dark[4]),
        },
        '.recharts-cartesian-grid line[stroke="#ccc"]': {
            stroke: onlyIf(dark, colors.gray[6]),
        }
    }} />

}
```

Overall, Tailwind colors (aka what they use) match up quite nicely with mantine. From what I've found, manually overriding all their colors (especially their svg fills) is somewhat of a pain, and honestly, it looks fine when in dark mode. I made sure to use the same background and border colors as Mantine's [Paper](https://mantine.dev/core/paper/) component in dark mode.

![chrome_jV70nSSE7A](https://user-images.githubusercontent.com/54898870/214354956-b1b98261-391a-4182-b957-142113a925e6.gif)

**Reply by dukesx** [reply]
Thanks mate!

**Reply by Manas-Nagelia** [reply]
This is great. Hopefully Mantine can natively support it in their library one day

**Reply by MathRobin** [reply]
Now done ;) https://mantine.dev/charts/getting-started/

**Reply by cseitz** [reply]
Love it!!!

**Reply by Manas-Nagelia** [reply]
Awesome! Looks great

**Comment by masterbater** [comment]
How to add annotations like chartjs https://www.chartjs.org/chartjs-plugin-annotation/latest/guide/types/box.html
